tests/ga/test_extension.py

```diff
@@ -1171,11 +1195,121 @@ def test_get_ext_handling_status(self, *args):
+
+        self.assertEqual(len(ext_status.substatusList), 1)
+        sub_status = ext_status.substatusList[0]
+        self.assertEqual(sub_status.code, "0")
+        self.assertEqual(sub_status.message, None)
+        self.assertEqual(sub_status.status, VALID_EXTENSION_STATUS.success)
+
+    @patch("azurelinuxagent.common.conf.get_lib_dir")
+    def test_collect_ext_status_very_large_status_message(self, mock_lib_dir, *args):
+        """
+        Testing collect_ext_status() with a very large status file (>128K) to see if it correctly parses the status
+        without generating a really large message.
+        """
+        ext_handler_i, extension = self._setup_extension_for_validating_collect_ext_status(mock_lib_dir,
+                                                                                "sample-status-very-large.json", *args)
+        ext_status = ext_handler_i.collect_ext_status(extension)
+
+        self.assertEqual(ext_status.code, 1)
+        self.assertEqual(ext_status.configurationAppliedTime, None)
+        self.assertEqual(ext_status.operation, "Enable")
+        self.assertEqual(ext_status.sequenceNumber, 0)
+        self.assertEqual(ext_status.message, "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Vestibulum non "
+                                             "lacinia urna, sit amet venenatis orci. Praesent maximus erat et augue "
+                                             "tincidunt, quis fringilla urna mollis. Fusce id lacus veli ... "
+                                             "[TRUNCATED]")
+        self.assertEqual(ext_status.status, VALID_EXTENSION_STATUS.success)
+        self.assertEqual(len(ext_status.substatusList), 1)
+        for sub_status in ext_status.substatusList:
+            self.assertEqual('[{"status": {"status": "success", "code": "1", "snapshotInfo": [{"snapshotUri": "https://md-rvr1sst1m0s0.blob.core.windows.net/p3w4cdkggwwl/abcd?snapshot=2019-06-06T23:53:14.9090608Z", "errorMessage": ... [TRUNCATED]', sub_status.name)
+            self.assertEqual(0, sub_status.code)
+            self.assertEqual('Lorem ipsum dolor sit amet, consectetur adipiscing elit. Vestibulum non lacinia urna, sit amet venenatis orci. Praesent maximus erat et augue tincidunt, quis fringilla urna mollis. Fusce id lacus veli ... [TRUNCATED]', sub_status.message)
+            self.assertEqual(VALID_EXTENSION_STATUS.success, sub_status.status)
+
+    @patch("azurelinuxagent.common.conf.get_lib_dir")
+    def test_collect_ext_status_read_file_read_exceptions(self, mock_lib_dir, *args):
+        """
+        Testing collect_ext_status to validate the readfile exceptions.
+        """
+        ext_handler_i, extension = self._setup_extension_for_validating_collect_ext_status(mock_lib_dir,
+                                                                                           "sample-status.json", *args)
+
+        with patch('azurelinuxagent.common.utils.fileutil.read_file') as patch_read_file:
```

Comment:
let's do this patch conditional so that it applies only to the status file. If the implementation of collect_ext_status ever needs to read another file likely this test will break.

For example, see #1800 

We need to be careful when mocking "basic" functions such as reading a file, logging a message, producing telemetry, etc